raiden/network/rpc/client.py

```diff
@@ -651,20 +662,39 @@ def poll(self, transaction_hash: TransactionHash) -> Dict[str, Any]:
         transaction_hash_hex = encode_hex(transaction_hash)
 
         while True:
-            # Returns `None` while the transaction isn't yet mined. A
-            # transaction is not guaranteed to be mined until the confirmation
-            # block is reached, because of this it is possible for the receipt
-            # to be set on one interation and for it to disappear on another
-            # (assuming a properly chosen confirmation_block number).
             tx_receipt = self.web3.eth.getTransactionReceipt(transaction_hash_hex)
 
-            if tx_receipt:
+            # Parity (as of 2.5.7) always returns a receipt. When the
+            # transaction is not mined in the canonical chain, the receipt will
+            # not have meaningful values. Example of receipt for a transaction
+            # that is not mined:
+            #
+            #   blockHash: None
+            #   blockNumber: None
+            #   contractAddress: None
+            #   cumulativeGasUsed: The transaction's gas
+            #   from: None
+            #   gasUsed: The transaction's gas
+            #   logs: []
+            #   logsBloom: Zero is hex
+            #   root: None
+            #   status: 1
+            #   to: None
+            #   transactionHash: The transaction's hash
+            #   transactionIndex: 0
+            #
+            # Geth only returns a receipt if the transaction was mined on the
+            # canonical chain. https://github.com/raiden-network/raiden/issues/4529
+            is_transaction_mined = tx_receipt and tx_receipt.get("blockNumber") is not None
+
+            if is_transaction_mined:
                 confirmation_block = (
```

Comment:
Without the check `tx_receipt.get("blockNumber") is not None` above, here `tx_receipt["blockNumber"] + self.default_block_num_confirmations` we would have something like `None + 5` which raises an exception.